earth_enterprise/src/third_party/skia/SConscript

```diff
@@ -18,29 +18,57 @@
 
 import time
 import os
+import subprocess
+import re
 Import('third_party_env')
 
 current_dir = Dir('.').abspath
 skia_version = 'skia-5.1.2'
 
+# Gather the listing of skia source files from the tarball
+skia_source = File('#/../../earth_enterprise/third_party/skia/%s.tar.gz'
+                   % skia_version).abspath
+
+# This regexp selects the desired groups of cpp files out from everything else.
+# Strings are assumed to begin with './' as they are in the skia-5.1.2 tarball,
+# and may have trailing whitespace (e.g. newlines).  The path of the cpp file
+# is captured.
+cpp_select_re = re.compile(r'^\./(src/(?:core|effects|effects/gradients|gpu|gpu/gl|gpu/effects|images|image|pathops|utils)/[^/]+\.cpp)\s*$')
+# This regexp identifies certain cpp files to be removed from the list selected by cpp_select_re
+cpp_downselect_re = re.compile(r'^src/utils/[^/]*(?:SkDeferredCanvas|SkLua|SkThreadUtils_pthread_mach|SkThreadUtils_win|SkDumpCanvas|SkCamera)[^/]*\.cpp$')
+
+# Get a listing of the contents of the skia tarball
+tar_p = subprocess.Popen(['tar', 'tzf', skia_source], stdout=subprocess.PIPE)
```

Comment:
In looking through some more of the third-party libraries, it looks like there aren't too many places where surveying the contents of the tarball beforehand will help us. We'll keep the technique in our pocket, though.

Reply:
Because we are using fake output files (the dot files) to represent the output of the tar operation all over third party scons can not properly really know if the tar file needs to be re-extracted.  If some of all of the source is deleted for whatever reason but scons sees that dot file it will think executing the un-tar command is not needed because all if the output exists.  The more you tell scons the better and more accurate job it can do.